backend/taco_backend.py

```diff
@@ -18,6 +18,43 @@
 import sparse
 from .core import Backend
 
+# Hack for @ operator for simplicity. No dimension checks.
+setattr(pt.tensor, '__matmul__', lambda x, y: TacoBackend.dot(x, y))
+
+
+def _scalarfunc(t):
+    if len(t.shape) == 0:
+        # TODO(yejiayu): Extend this to support any type casts.
+        return float(t[0])
+    else:
+        raise TypeError("Only rank 0 can be converted to scalars.")
+
+
+setattr(pt.tensor, '__float__', _scalarfunc)
+
+###############################################################################
+# Impl to get around pytaco slicing.
+###############################################################################
+__pt_impl_get_item = getattr(pt.tensor, '__getitem__')
+
+
+def _getitem(self, index):
+    if index is None or isinstance(index, int) or len(index) == 0:
+        return __pt_impl_get_item(self, index)
+
+    for i in index:
+        if isinstance(i, pt.core.core_modules.index_var):
+            return __pt_impl_get_item(self, index)
```

Comment:
is the second argument i rather than index? also what does index represent for this case?

Reply:
(It's possible that this part is not well tested, we can check codecov later).

So the idea is pytaco support int or list of int for slicing in general. And it also support index_vars to pass in `__getitem__`. The latter case, index vars are mostly for index expressions which we are not touching upon (though it will be used in pt.einsum for example).

Here we are trying to say, if there is an IndexVar in the index array, we just call the original implementation with the **same** argument (hence index).

Reply:
I see